### PR TITLE
Adding copy reports script

### PR DESCRIPTION
**Pull request by williamkubecost**
This is a script to migrate over reports/budgets to V2

### CODE REVIEW

File: copy-reports/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copy Reports Script
```
Comment (ameijer) [inline copy-reports/README.md:1]: also copies budgets

File: copy-reports/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+## Usage
+
+1. Open a terminal.
+
+2. Copy and paste the script into a file, e.g., `copy-reports.sh`.
+
+3. Make the script executable:
+   ```bash
+   chmod +x copy-reports.sh
+   ./copy-reports.sh
```
Comment (ameijer) [inline copy-reports/README.md:15]: to keep it simpler, you can always just use the shell to execute it as a readable file:

```
bash copy-reports.sh
```
then no need to set the execute bit

File: copy-reports/copy-reports.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+OLD_POD=$(kubectl get pod -n kubecost -l app=cost-analyzer -o jsonpath='{.items[0].metadata.name}')
```
Comment (ameijer) [inline copy-reports/copy-reports.sh:3]: take namespace as a parameter and blow the user up if they dont provide it

File: copy-reports/copy-reports.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+NEW_POD=$(kubectl get pod -n kubecost -l app=aggregator -o jsonpath='{.items[0].metadata.name}')
```
Comment (ameijer) [inline copy-reports/copy-reports.sh:4]: I would add a check here that ensures their aggregator is deployed as a stateful set, as a guardrail

File: copy-reports/copy-reports.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "cloud-cost-reports.json"
+    "reports.json"
+    "alerts/alerts.json"
+)
+
+for FILE in "${FILES[@]}"; do
+    echo "Copying file: $FILE"
+
+    # Check the file in the old reports
+    kubectl exec -it -n kubecost $OLD_POD -c cost-model -- ls -lh "$SOURCE_DIR/$FILE"
```
Comment (ameijer) [inline copy-reports/copy-reports.sh:20]: parameterized namespace

File: copy-reports/copy-reports.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (ameijer) [inline copy-reports/copy-reports.sh:1]: set -ex

File: copy-reports/copy-reports.sh (hunk 5 above)
Comment (ameijer) [reply copy-reports/copy-reports.sh:20]: do you do anything with this ls output? is this command necessary?

File: copy-reports/copy-reports.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    echo "Copying file: $FILE"
+
+    # Check the file in the old reports
+    kubectl exec -it -n kubecost $OLD_POD -c cost-model -- ls -lh "$SOURCE_DIR/$FILE"
+
+    # Download the file from the old reports
+    kubectl cp kubecost/$OLD_POD:"$SOURCE_DIR/$FILE" /tmp/"$FILE" -c cost-model
+
+    # Upload the file to the new reports location in Aggregator
+    kubectl cp /tmp/"$FILE" kubecost/$NEW_POD:"$TARGET_DIR/$FILE" -c aggregator
```
Comment (ameijer) [inline copy-reports/copy-reports.sh:26]: does kubectl cp work to copy directly from one pod to a new pod? or must you go to your local? never tried direct pod to pod xfer

**Review by ameijer — COMMENTED** [review]
see what you think about these ideas. feel free to re-request my review so it emails me when you want more eyes

File: copy-reports/copy-reports.sh (hunk 5 above)
Comment (williamkubecost) [reply copy-reports/copy-reports.sh:20]: this is probably not necessary, we know exactly which files need to be copied over.

File: copy-reports/copy-reports.sh (hunk 7 above)
Comment (williamkubecost) [reply copy-reports/copy-reports.sh:26]: don't believe you copy from pod to pod, only to local.

File: copy-reports/copy-reports.sh [hunk 8]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -ex
+set -euo pipefail
```
Comment (ameijer) [inline copy-reports/copy-reports.sh:3]: you can always do euxo to just get this on one line. if you are on the fence about x, dont use it, but I think scripts people run at home, it is always nice to show them what you are doing

Comment (michaelmdresser) [reply copy-reports/copy-reports.sh:3]: ^^^^ absolutely
`set -euxo pipefail`

File: copy-reports/copy-reports.sh [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+fi
+
+# Check if the provided namespace exists
+if ! kubectl get namespace "$NAMESPACE" &>/dev/null; then
+    echo "Namespace '$NAMESPACE' does not exist. Exiting."
+    exit 1
+fi
+
+# Check if aggregator is deployed as a statefulset
+if helm get values kubecost -n $NAMESPACE | grep -q "deployMethod: statefulset"; then
```
Comment (ameijer) [inline copy-reports/copy-reports.sh:19]: I like the approach - however, you would then need the user to provide the name of their release as a parameter as well. Here's a bit of context that might help us simplify this: 

> In a kubecost deployment, there is only one time that a statefulset with the label `app: aggregator` is deployed - when kubecost is running in aggregator statefulset mode

So, you could get sts with the label app: aggregator in the kubecost namespace, which we already asked the user for. If no results returned, then you fail the guard. for example: 
```
$ kubectl -n alex get sts -l app=aggregator
NAME                 READY   AGE
kc-alex-aggregator   1/1     4h46m

 kubectl -n kube-system  get sts -l app=aggregator
No resources found in kube-system namespace.
```

**Review by ameijer — COMMENTED** [review]
one last ask will, then we can ship (at least from my perspective)

**Review by ameijer — APPROVED** [review]
LGTM

**Pull request merged**